scipy/stats/tests/test_stats.py

```diff
@@ -1637,6 +1637,42 @@ def test_mad(self):
         mad_expected = np.asarray([0.644931, 0.7413, 0.66717, 0.59304])
         assert_array_almost_equal(mad, mad_expected)
 
+    def test_mad_empty(self):
+        dat = []
+        mad = stats.median_absolute_deviation(dat)
+        assert_equal(mad, np.nan)
+
+    def test_mad_empty(self):
+        dat = []
+        mad = stats.median_absolute_deviation(dat)
+        assert_equal(mad, np.nan)
+
+    def test_mad_nan_propagate(self):
+        dat = np.array([2.20, 2.20, 2.4, 2.4, 2.5, 2.7, 2.8, 2.9, 3.03,
+                3.03, 3.10, 3.37, 3.4, 3.4, 3.4, 3.5, 3.6, 3.7, 3.7,
+                3.7, 3.7,3.77, 5.28, np.nan])
+
+        mad = stats.median_absolute_deviation(dat, nan_policy='propagate')
+        assert_equal(mad, np.nan)
+
+    def test_mad_nan_raise(self):
+        dat = np.array([2.20, 2.20, 2.4, 2.4, 2.5, 2.7, 2.8, 2.9, 3.03,
+                3.03, 3.10, 3.37, 3.4, 3.4, 3.4, 3.5, 3.6, 3.7, 3.7,
+                3.7, 3.7,3.77, 5.28, np.nan])
+
+        with assert_raises(ValueError):
+            mad = stats.median_absolute_deviation(dat, nan_policy='propagate')
+
+    def test_mad_nan_omit(self):
+        dat = np.array([2.20, 2.20, 2.4, 2.4, 2.5, 2.7, 2.8, 2.9, 3.03,
+                3.03, 3.10, 3.37, 3.4, 3.4, 3.4, 3.5, 3.6, 3.7, 3.7,
+                3.7, 3.7,3.77, 5.28, np.nan])
+
+        mad = stats.median_absolute_deviation(dat, nan_policy='omit')
+        assert_almost_equal(mad, 0.504084)
+
+
+
```

Comment:
Only need 2 blank lines between classes, this third line causes a pycodestyle error on the CI builds, please delete the 3rd blank line.